src/main/java/mate/academy/springbootintro/service/BookServiceImpl.java

```diff
@@ -17,27 +21,49 @@ public class BookServiceImpl implements BookService {
 
     private final BookRepository bookRepository;
     private final BookMapper bookMapper;
+    private final BookSpecificationBuilder bookSpecificationBuilder;
 
     @Override
     public BookDto createBook(CreateBookRequestDto createBookRequestDto) {
         Book book = bookMapper.toModel(createBookRequestDto);
         book.setIsbn("abc" + new Random().nextInt(1000));

```

Comment:
it was overlooked in previous reviews, but you should make this `Random` instance a constant.